isettings.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2018-2021 QuasarApp.
+ * Distributed under the lgplv3 software license, see the accompanying
+ * Everyone is permitted to copy and distribute verbatim copies
+ * of this license document, but changing it is not allowed.
+*/
+
+#include "isettings.h"
+#include <QSettings>
+#include <QCoreApplication>
+
+using namespace QuasarAppUtils;
+
+ISettings::ISettings(SettingsSaveMode mode) {
+    _mode = mode;
+}
+
+SettingsSaveMode ISettings::getMode() const {
+    return _mode;
+}
+
+void ISettings::setMode(const SettingsSaveMode &mode) {
+    _mode = mode;
+}
+
+//ISettings *ISettings::initSettings(SettingsSaveMode mode) {
+//    static ISettings* res = new ISettings(mode);
+//    return res;
+//}
```

Comment:
* remove this code 
* create a one private static pointer to ISettings class.
* create a template  method on header that will be named as **instance**
this method should be check incoming type if is base of ISettings and check your private pointer if null
if type is base of I Settings and pointer is null the you need to create new object with incomming template type.
* return your static pointer

Reply:
example of validation template type :
``` cpp
        static_assert (std::is_base_of<TemplateType, CompareType>::value,
                "the TemplateType type must be CompareType");
```